delivery/client.py

```diff
@@ -109,9 +123,93 @@ class DeliveryServiceClient:
     def __init__(
         self,
         routes: DeliveryServiceRoutes,
+        cfg_factory=None,
     ):
         self._routes = routes
+        self.cfg_factory = cfg_factory or ci.util.ctx().cfg_factory()
         self.session = requests.sessions.Session()
+        self.bearer_token = None
+
+    def _authenticate(self):
+        if self.bearer_token:
```

Comment:
nit: should not we also check whether jwt-token is still valid (not expired) and re-authenticate if this is the case?